tests/MongoDBDriver_APM-001.phpt

```diff
@@ -0,0 +1,578 @@
+--TEST--
+HHVM-190: APM Specification
+--FILE--
+<?php
+include dirname(__FILE__) . '/utils.inc';
+
+$m = new MongoDB\Driver\Manager("mongodb://localhost:27017");
+cleanup( $m );
+
+class MySubscriber implements MongoDB\Driver\Monitoring\CommandSubscriber
+{
+	public function commandStarted( \MongoDB\Driver\Monitoring\CommandStartedEvent $event )
+	{
+		echo "started:\n";
+		var_dump( $event );
+	}
+
+	public function commandSucceeded( \MongoDB\Driver\Monitoring\CommandSucceededEvent $event )
+	{
+		echo "succeeded:\n";
+		var_dump( $event );
+	}
+
+	public function commandFailed( \MongoDB\Driver\Monitoring\CommandFailedEvent $event )
+	{
+		echo "failed:\n";
+		var_dump( $event );
+	}
+}
+
+$m->addSubscriber( new MySubscriber() );
+cleanup( $m );
+
+$d = new \MongoDB\BSON\Decimal128("1234.5678");
+
+$bw = new MongoDB\Driver\BulkWrite( [ 'ordered' => false ] );
+$_id = $bw->insert( [ 'decimal' => $d ] );
+$r = $m->executeBulkWrite( 'demo.test', $bw );
+
+$query = new MongoDB\Driver\Query( [] );
+$cursor = $m->executeQuery( "demo.test", $query );
+var_dump( $cursor->toArray() );
+?>
+--EXPECTF--
+started:
+object(MongoDB\Driver\Monitoring\CommandStartedEvent)#8 (6) {
+  ["command":"MongoDB\Driver\Monitoring\CommandStartedEvent":private]=>
+  object(stdClass)#10 (1) {
+    ["drop"]=>
+    string(4) "test"
+  }
+  ["databaseName":"MongoDB\Driver\Monitoring\CommandStartedEvent":private]=>
+  string(4) "demo"
+  ["commandName":"MongoDB\Driver\Monitoring\_CommandEvent":private]=>
+  string(4) "drop"
+  ["server":"MongoDB\Driver\Monitoring\_CommandEvent":private]=>
+  object(MongoDB\Driver\Server)#9 (10) {
+    ["host"]=>
+    string(9) "localhost"
+    ["port"]=>
+    int(27017)
+    ["type"]=>
+    int(1)
+    ["is_primary"]=>
+    bool(false)
+    ["is_secondary"]=>
+    bool(false)
+    ["is_arbiter"]=>
+    bool(false)
+    ["is_hidden"]=>
+    bool(false)
+    ["is_passive"]=>
+    bool(false)
+    ["last_is_master"]=>
+    array(9) {
```

Comment:
Based on other tests, you likely want to use a wildcard pattern here and avoid asserting the `isMaster` response document. You may even want to wildcard the entire Server.